### PR TITLE
[Reviewer: Rob] SNMP fixes

### LINKED ISSUE #76: alarms_agent doesn't log to syslog
I couldn't see any logs generated by the alarm_agent (specifically alarm_req_listener.cpp) and noticed that the snmp subagent start up (alarms_agent.cpp) was not enabling syslog, so I enabled it as follows:
```
@ -75,11 +75,11 @@ int main (int argc, char **argv)
         abort ();
       }
   }
 
   // Log SNMP library output to syslog
-  snmp_enable_calllog();
+  snmp_enable_syslog_ident("alarms_agent", LOG_DAEMON);
 
   // Set ourselves up as a subagent
   netsnmp_ds_set_boolean(NETSNMP_DS_APPLICATION_ID, NETSNMP_DS_AGENT_ROLE, 1);
   init_agent("clearwater-alarms");
 ```

### LINKED ISSUE #77: alarm_req_listener is not handling zmq_bind error
When alarm_req_listener tries to zmq_bind to /var/run/clearwater/alarms and it fails, no action is taken so there are conditions underwhich alarms don't work.

I fixed this by exiting the alarms_agent upon encountering such an error as follows:
```
@@ -158,9 +158,14 @@ bool AlarmReqListener::zmq_init_sck()
 
   if (rc == -1)
   {
-    snmp_log(LOG_ERR, "zmq_bind failed: %s", zmq_strerror(errno));
+    // LCOV_EXCL_START
+    snmp_log(LOG_ERR, "zmq_bind(.., %s) failed: %s - killing myself", sck_url.c_str(), zmq_strerror(errno));
+    kill(getpid(), SIGKILL);
+    // LCOV_EXCL_STOP
     return false;
   }
+  
+  snmp_log(LOG_INFO, "zmq_bind(.., %s) succeeded", sck_url.c_str());
 
   rc=chmod(sck_file.c_str(), 0777);
   if (rc == -1)
```
I resorted to killing because past experience with net-snmp hanging on exit has left me weary of allowing any exit handlers to run.

### LINKED ISSUE #78: alarms_agent.cpp is sending traps to the wrong port
Traps should be sent to port 162:
```
@@ -88,7 +88,7 @@ int main (int argc, char **argv)
        ii != trap_ips.end();
        ii++)
   {
-    create_trap_session(const_cast<char*>(ii->c_str()), 161, community,
+    create_trap_session(const_cast<char*>(ii->c_str()), 162, community,
                         SNMP_VERSION_2c, SNMP_MSG_INFORM);  
   }
```

### PR DESCRIPTION
**Pull request by eleanor-merry**
Rob, can you review these fixes for the alarms agent. The main fix is that we didn't check the output of running start on the AlarmReqListener, so we could run a broken process

Tested live

Fixes #78, fixes #77, fixes #76

### CODE REVIEW

File: alarm_req_listener.cpp [hunk 1]
```diff
@@ -166,6 +161,9 @@ bool AlarmReqListener::zmq_init_sck()
   if (rc == -1)
   {
     snmp_log(LOG_ERR, "chmod(%s, 0777) failed: %s", sck_file.c_str(), strerror(errno));
+#ifndef UNIT_TEST
+    return false;
```
Comment (rkd-msw) [inline alarm_req_listener.cpp]: Why not return false in UT?

File: alarms_agent.cpp [hunk 2]
```diff
@@ -78,6 +78,7 @@ int main (int argc, char **argv)
 
   // Log SNMP library output to syslog
```
Comment (rkd-msw) [inline alarms_agent.cpp]: Should we be logging to syslog, or should we be creating /var/log/clearwater-alarms/ and logging there?

**Comment by rkd-msw** [comment]
Looks good. One question, one WIBNI.

File: alarms_agent.cpp (hunk 2 above)
Comment (eleanor-merry) [inline alarms_agent.cpp]: Perhaps, but I'm not going to do so in this PR

**Pull request merged**